src/main/java/net/ckozak/ExecutorBenchmarks.java

```diff
@@ -199,20 +204,72 @@ public void submitBurst(ExecutorService executor)
 
         public void awaitBurstCompletion() throws ExecutionException, InterruptedException
         {
+            // with spinWaitCompletion it attempt to check completion of previous tasks
+            // while awaiting the current task to complete
+            boolean spinWaitCompletion = this.spinWaitCompletion;
             Future<?>[] pendingTasks = this.pendingTasks;
+            int lowerIndexToCheck = 0;
             for (int i = 0, size = burstSize; i < size; i++) {
                 int index = size - (i+1);
+                if (index < lowerIndexToCheck)
+                {
+                    break;
+                }
                 Future<?> lastTask = pendingTasks[index];
-                pendingTasks[index] = null;
-                lastTask.get();
+                if (lastTask != null)
+                {
+                    pendingTasks[index] = null;
+                    if (!spinWaitCompletion)
+                    {
+                        lastTask.get();
+                    }
+                    else
+                    {
+                        while (!lastTask.isDone())
+                        {
+                            final int remaining = index - lowerIndexToCheck;
+                            if (remaining > 0)
+                            {
+                                lowerIndexToCheck = deleteCompleted(pendingTasks, lowerIndexToCheck, remaining);
+                            } else {
+                                /* TODO: evaluate the fairness of this: it should happen only when
+                                         this task is the very last one */
+                                // Thread::onSpinWait() is a better candidate
+                                Thread.yield();
```

Comment:
I'll update this project to use java 11 and switch this to onSpinWait, I don't think yield is ideal in a potentially tight loop, and could impact results in unexpected ways.